### ISSUE TITLE
TR1X bug: Passport shifts after opening animation when using quicksave/quickload

### ISSUE THREAD

**Issue by aredfan** [LostArtefacts/TRX#1143]
When the passport is almost open, the object shifts into position a little up and to the right.  I've confirmed that this bug is a regression since 2.15.  The bug is not present in 2.14, nor in TombATI.

To see the bug more clearly, I would advise reducing UI text scale to 0.5, then observe the passport's opening animation after pressing F6.

@lahm86 tagging you because you've investigated something like this before in #928.  I think this issue could be related?

![20240119_222346_Caves](https://github.com/LostArtefacts/TR1X/assets/94776985/b4da705f-d239-4ab8-9f77-66a98aaf2eb6)

**Comment by lahm86** [comment]
It seems to be the same for me in TombATI, but I could be looking at it wrongly. If you open a level in WadTool and go into object 71's animations, you'll see the slight shift between frames 13 and 14. Frame 14 is where the passport animation stops initially. Is this the shift you mean?

**Comment by aredfan** [comment]
I'll show you what I mean, I made a comparison video to help illustrate the issue.

Left is TombATI, right is TR1X

https://github.com/LostArtefacts/TR1X/assets/94776985/4311ca85-7543-49ba-968f-a419b9376225

There is a little discrepancy between the two videos even if I tried to sync them up.  At the end of the passport opening animation, the shift is more evident in TR1X.

**Comment by lahm86** [comment]
Thanks, I do see it now. I managed to get a similar recording with the animation lined up - you can see in TR1X at 0:25 that it pauses for an extra frame for some reason.
https://youtu.be/7saXw3s3il4

Nothing is jumping out at me in the code so far. Any ideas @rr, @walkawayy?

**Comment by walkawayy** [comment]
Hm I thought maybe OG ended one frame early. But the frame timings are synced until this divergence. It's like TR1X is stuck for a frame and holds this frame for one extra. After this frame, their timings are synced again but TR1X is a frame behind. Then it skips the last movement frame to reach the end of the animation.

![image](https://github.com/LostArtefacts/TR1X/assets/81546780/6e20deb6-2caa-43d3-9d44-e85ee67c66dc)

Definitely a bug, but it could be long, ugly debugging for a tiny one frame bug. Unless it is affecting more. I'd guess it's something slightly different in the new passport ordering code.

**Comment by Richard-L** [comment]
Impressive find @aredfan

**Comment by walkawayy** [comment]
@lahm86 I looked through 2.15 and hotfix patches and nothing stood out. Debug prints look fine. Idk.

```
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 0; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 1; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 1; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 2; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 2; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 3; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 3; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 4; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 4; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 5; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 5; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 6; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 6; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 7; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 7; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 8; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 8; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 9; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 9; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 10; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 10; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 11; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 11; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 12; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 12; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 13; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 13; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 14; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
```

**Comment by aredfan** [comment]
I made recordings of the quickload animation from previous builds, if applicable, and inspected the videos frame-by-frame.  I also slowed down the speed of my videos to 25%.  I put my findings in this spreadsheet: 

https://docs.google.com/spreadsheets/d/18tZeBMZdNwWeVJGlIiVvZ6FPAphES8Vli1mpwKYjFks/edit?usp=sharing

It seems the first and second animation frames appear for twice as long when compared to TombATI.  This could be the cause of the passport getting stuck/shifting.

**Comment by walkawayy** [comment]
Just to clarify, is that from 2.15, or 2.15.3?

So your chart is saying it's actually the first two frames that are displayed longer? And is this only a problem for quick load, or normal passport opening too like title screen?

**Comment by aredfan** [comment]
Yep 2.15, from here: https://github.com/LostArtefacts/TR1X/releases/tag/2.15

The first two animation frames last longer when compared to TombATI but I'm not sure if it's related.  One thing that is consistent is the quickload passport seem to get stuck on animation frame 37.

I've not seen the passport shifting in other cases like the main menu.

**Comment by walkawayy** [comment]
Ah ok. My post above was a test from the main menu. So maybe it's limited to something with quick load / quick save.

**Comment by walkawayy** [comment]
Hm quick load seems to look the same:

3.1:
```
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 0; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 1; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 1; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 2; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 2; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 3; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 3; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 4; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 4; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 5; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 5; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 6; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 6; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 7; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 7; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 8; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 8; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 9; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 9; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 10; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 10; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 11; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 11; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 12; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 12; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 13; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 13; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 224 Inv_AnimateItem current_frame: 14; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
```


2.15:
```
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 0; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 1; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 1; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 2; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 2; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 3; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 3; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 4; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 4; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 5; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 5; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 6; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 6; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 7; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 7; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 8; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 8; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 9; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 9; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 10; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 10; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 11; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 11; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 12; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 12; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 13; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 13; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../src/game/inventory/inventory.c 219 Inv_AnimateItem current_frame: 14; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
```

**Comment by walkawayy** [comment]
Idk what's causing this but I think I'm done hunting. `Inv_ConstructAndDisplay` seems fine, so the delay must be coming from somewhere else?

```
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 0; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 1; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 2; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 3; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 4; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 5; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 6; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 7; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 8; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 9; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 10; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 11; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 12; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 13; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 0
../../src/game/option/option_passport.c 614 Option_Passport current_frame: 14; frames_total: 30; goal_frame: 14; open_frame: 14; 
../../src/game/inventory/inventory.c 759 Inv_ConstructAndDisplay current_frame: 14; frames_total: 30; goal_frame: 14; open_frame: 14; anim_count: 1
```

**Comment by aredfan** [comment]
I just checked and apparently this issue is fixed in a recent snapshot build after #1315.  The bug was still present in 4.0.3 when playing in 30 FPS mode.  I'm not sure what's changed in the code that fixed this issue but it's good news.  Thank you.
